akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -393,62 +396,19 @@ class ShardCoordinator(typeName: String, settings: ClusterShardingSettings,
     Cluster(context.system).unsubscribe(self)
   }
 
-  override def receiveRecover: Receive = {
-    case evt: DomainEvent ⇒
-      log.debug("receiveRecover {}", evt)
-      evt match {
-        case ShardRegionRegistered(region) ⇒
-          persistentState = persistentState.updated(evt)
-        case ShardRegionProxyRegistered(proxy) ⇒
-          persistentState = persistentState.updated(evt)
-        case ShardRegionTerminated(region) ⇒
-          if (persistentState.regions.contains(region))
-            persistentState = persistentState.updated(evt)
-          else {
-            log.debug("ShardRegionTerminated, but region {} was not registered. This inconsistency is due to that " +
-              " some stored ActorRef in Akka v2.3.0 and v2.3.1 did not contain full address information. It will be " +
-              "removed by later watch.", region)
-          }
-        case ShardRegionProxyTerminated(proxy) ⇒
-          if (persistentState.regionProxies.contains(proxy))
-            persistentState = persistentState.updated(evt)
-        case ShardHomeAllocated(shard, region) ⇒
-          persistentState = persistentState.updated(evt)
-        case _: ShardHomeDeallocated ⇒
-          persistentState = persistentState.updated(evt)
-      }
-
-    case SnapshotOffer(_, state: State) ⇒
-      log.debug("receiveRecover SnapshotOffer {}", state)
-      //Old versions of the state object may not have unallocatedShard set,
-      // thus it will be null.
-      if (state.unallocatedShards == null)
-        persistentState = state.copy(unallocatedShards = Set.empty)
-      else
-        persistentState = state
-
-    case RecoveryCompleted ⇒
-      persistentState.regionProxies.foreach(context.watch)
-      persistentState.regions.foreach { case (a, _) ⇒ context.watch(a) }
-      persistentState.shards.foreach { case (a, r) ⇒ sendHostShardMsg(a, r) }
-      allocateShardHomes()
-  }
-
-  override def receiveCommand: Receive = {
+  def active: Receive = {
     case Register(region) ⇒
       log.debug("ShardRegion registered: [{}]", region)
       aliveRegions += region
-      if (persistentState.regions.contains(region))
+      if (state.regions.contains(region))
         sender() ! RegisterAck(self)
       else {
         gracefulShutdownInProgress -= region
-        saveSnapshotWhenNeeded()
-        persist(ShardRegionRegistered(region)) { evt ⇒
-          val firstRegion = persistentState.regions.isEmpty
-
-          persistentState = persistentState.updated(evt)
+        update(ShardRegionRegistered(region)) { evt ⇒
+          val firstRegion = state.regions.isEmpty
+          state = state.updated(evt)
           context.watch(region)
-          sender() ! RegisterAck(self)
+          region ! RegisterAck(self)
```

Comment:
sender() and region is the same, right?
but now we use region here, but sender() a few lines above